src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+public abstract class Figure implements AreaCalculator, DrawGenerator {
```

Comment:
The class name 'Figure' is appropriate as it is a noun. However, ensure that the interfaces 'AreaCalculator' and 'DrawGenerator' are named correctly according to their responsibilities and do not include verbs.